antarest/study/service.py

```diff
@@ -1896,15 +1897,14 @@ def _save_study(
         if isinstance(study, RawStudy):
             study.content_status = content_status
 
-        if owner:
-            study.owner = self.user_service.get_user(owner.impersonator, params=RequestParameters(user=owner))
-            groups = []
-            for gid in group_ids:
-                group = next(filter(lambda g: g.id == gid, owner.groups), None)
-                if group is None or not group.role.is_higher_or_equals(RoleType.WRITER) and not owner.is_site_admin():
-                    raise UserHasNotPermissionError()
-                groups.append(Group(id=group.id, name=group.name))
-            study.groups = groups
+        study.owner = self.user_service.get_user(owner.impersonator, params=RequestParameters(user=owner))
+        groups = []
+        for gid in group_ids:
+            group = next(filter(lambda g: g.id == gid, owner.groups), None)
+            if group is None or not group.role.is_higher_or_equals(RoleType.WRITER) and not owner.is_site_admin():
+                raise UserHasNotPermissionError()
```

Comment:
Add an error message. This is a "Permission denied" error.
You should also add an error message in the previous `raise UserHasNotPermissionError`.